lib/syskit/models/port.rb

```diff
@@ -105,19 +105,13 @@ def to_component_port
             # @raise [SelfConnection]
             def connect_to(in_port, policy = {})
                 out_port = to_component_port
+                if out_port.respond_to?(:recommend_init)
+                    policy[:init] = out_port.recommend_init
```

Comment:
You shouldn't be setting this if `init` was given explicitly by the caller. And you really shouldn't be changing the argument (i.e. should be `policy = policy.merge(init: out_port.recommend_init)`)